### ISSUE TITLE
[MVUX] Update Bindable Generator Attribute Naming

### ISSUE THREAD

**Issue by ajpinedam** [unoplatform/uno.extensions#2515]
## What would you like to be added:
Rename the Bindable Generator Attribute to something more generic, especially now that this is not generating BindableModels anymore but instead it is generating ViewModels

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renders for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks

**Comment by dr1rrb** [comment]
Considering that the [`BindableAttribute`](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.bindableattribute?view=net-8.0) is the attribute used by windows to flag object that should be accessible to the binding engine, I would not introduce a breaking change to rename those attributes.

**Comment by francoistanguay** [comment]
Didnt realize BindableGenerator was "associated" to the BindableAttribute idea. Happy to keep it as-is for now.

**Comment by DevTKSS** [comment]
My compiler is also telling that on each viewModel the Bindable attribute is missing although the mainmodel and MainPage having it added by me

**Comment by DevTKSS** [comment]
@ajpinedam 
`The Bindable attribute is missing and the type [UnoHotDesignApp1.Presentation.MainViewModel] is not known by the MetadataProvider. Reflection was used instead of the binding engine and generated static metadata. Add the Bindable attribute to prevent this message and performance issues.`

that is happening on mvux, where I essentially can not even add the BindableAttribuge it is telling me to do. if we would be able to get the generator add this attribute, that maybe could solve this....

**Comment by Mike-E-angelo** [comment]
I am seeing the same messages that @DevTKSS is seeing as well:

```
The Bindable attribute is missing and the type [<Root>.Application.Presentation.MainPage] is not known by the MetadataProvider. Reflection was used instead of the binding engine and generated static metadata. Add the Bindable attribute to prevent this message and performance issues.
The Bindable attribute is missing and the type [<Root>.Application.Presentation.MainViewModel] is not known by the MetadataProvider. Reflection was used instead of the binding engine and generated static metadata. Add the Bindable attribute to prevent this message and performance issues.
```

**Comment by dr1rrb** [comment]
@DevTKSS @Mike-E-angelo We are already [adding a `[BindableAttribute]` on VM generated by MVUX](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive.Generator/Bindables/ViewModelGenTool_3.cs#L101) which then should be considered by the [bindable type provider generator](https://github.com/unoplatform/uno/blob/master/src/SourceGenerators/Uno.UI.SourceGenerators/BindableTypeProviders/BindableTypeProvidersGenerationTask.cs#L133). So it seems to indicate that we have an another issue. On which platform do you get those error messages?

Also @Mike-E-angelo it's surprising that you also get this message for the `MainPage`. If it inherits from a `DepndencyObject` the attribute should be set transitively (as it's been set on `DependencyObject` by uno). What is the base type of your `MainPage`?

**Comment by Mike-E-angelo** [comment]
Hi @dr1rrb I am seeing this on `net9.0-desktop` (the only platform I am trying right now).  Additionally `MainPage` is a basic `Microsoft.UI.Xaml.Controls.Page, Uno.UI` page. 🤔

**Comment by DevTKSS** [comment]
@dr1rrb would have to check if its .net 8 or 9 but i am absolutely sure my current one is desktop only target because I am doing studio stuff where i had the issue before with a additional Windows target its really slow and way more buggy.
But the Bindableattribute related issue is happening at build even before i could even try to enter Studio.
I did get the issue Before (about sdk version 43 or so?) @Mike-E-angelo did tell about with the Mainpage also not having this attribute. It told it to be mainModel, mainPage and in that case not mentioning the viewModel 😅 when i added this attribute above the class it has been fine, only issue has been then also, the x:Bind to not working with mvux what I would have seen as the big advantage ob mvux to mvvm, and this would be (to me) a good reason to maybe also learn mvvm in the future.
In the application where this stopped i think i had been using desktop, android, ios, windows... You could look into my GitHub repos like the "MyUnoXamlApp" or similar named. Each of them is having issues why i did not continue to work on them. Not got mvux in my head then, starting with navigation which I currently carefully maybe have running in my current application.
But the downside, the last studio using application I committed and pushed to GitHub is after cloning to my local device not longer nicly running. Its hell to work with it. Laggy +1000%, login issue und studio not longer showing up (it did before!)
So for that reason I am currently porting the app comtent to the new one which is only targeting desktop and running smoothly. Okay with minor laggyness but way less than at the best time of the +windows one!

**Comment by dr1rrb** [comment]
Ok so if both of you are getting this issue on desktop target, it means it's definitely linked to one of our generators (MVUX or Uno)

But this is interesting! cc. @jeromelaban

> So I am noticing one of the warnings go away when I update my code-behind from this: `[../..]`

It means that it's probably only that MVUX's  `[Bindable]` is not considered by the generator ... or it's generated too late! Opened a new issue for that https://github.com/unoplatform/uno.extensions/issues/2648 (as it's not linked to the original issue).

**Comment by Youssef1313** [comment]
@dr1rrb This behavior is expected. https://github.com/unoplatform/uno.extensions/issues/2648#issuecomment-2574275998

**Comment by DevTKSS** [comment]
@Youssef1313 is this issue still current?

**Comment by dr1rrb** [comment]
No, closing it. We still have https://github.com/unoplatform/uno.extensions/issues/2648 to look at, but this is not linked to the original report.
